include/flatbuffers/flatbuffers.h

```diff
@@ -1606,22 +1616,16 @@ class FlatBufferBuilder {
   /// @param[in] len The number of bytes that should be stored from `str`.
   /// @return Returns the offset in the buffer where the string starts.
   Offset<String> CreateSharedString(const char *str, size_t len) {
-    FLATBUFFERS_ASSERT(FLATBUFFERS_GENERAL_HEAP_ALLOC_OK);
-    if (!string_pool)
-      string_pool = new StringOffsetMap(StringOffsetCompare(buf_));
-    auto size_before_string = buf_.size();
-    // Must first serialize the string, since the set is all offsets into
-    // buffer.
-    auto off = CreateString(str, len);
-    auto it = string_pool->find(off);
-    // If it exists we reuse existing serialized data!
-    if (it != string_pool->end()) {
-      // We can remove the string we serialized.
-      buf_.pop(buf_.size() - size_before_string);
-      return *it;
+    if (!string_pool) {
+      FLATBUFFERS_ASSERT(FLATBUFFERS_GENERAL_HEAP_ALLOC_OK);
+      string_pool = new StringPoolMap();
+      owned_string_pool = true;
     }
-    // Record this string for future use.
-    string_pool->insert(off);
+    auto s = std::string(str, len);
+    auto it = string_pool->find(s);
+    if (it != string_pool->end()) { return Offset<String>(it->second); }
+    auto off = CreateString(str, len);
+    string_pool->insert(std::make_pair(s, off.o));
```

Comment:
This probably created not just 1, but 2 `std::string` instances, unless we can massage this insertion to use `std::move` ?

Reply:
Yeah, std::move isn't really available to us if we have to be supportive of old standards.